great_expectations/core/factory/validation_definition_factory.py

```diff
@@ -103,3 +104,43 @@ def get(self, name: str) -> ValidationDefinition:
     def all(self) -> Iterable[ValidationDefinition]:
         """Get all ValidationDefinitions."""
         return self._store.get_all()
+
+    @public_api
+    def add_or_update(self, validation: ValidationDefinition) -> ValidationDefinition:
+        """Add or update an ValidationDefinition by name.
+
+        If an ValidationDefinition with the same name exists, overwrite it, otherwise
+        create a new ValidationDefinition.
+
+        Args:
+            validation: ValidationDefinition to add or update
+        """
+        try:
+            existing_validation = self.get(name=validation.name)
+        except DataContextError:
+            return self.add(validation=validation)
+
+        validation.id = existing_validation.id
+        self._add_or_update_batch_definition(validation)
+        self._add_or_update_suite(validation)
+        validation.save()
+
+        return validation
+
+    def _add_or_update_batch_definition(self, validation: ValidationDefinition) -> None:
+        data_source_manager = project_manager.get_data_source_manager()
+
+        batch_definition = validation.batch_definition
+        asset = batch_definition.data_asset
+        data_source = asset.datasource
```

Comment:
how do we expect a user to create these relationships?

Reply:
I don't think we do - we get these from the fluent API